cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
And what are your reasons? :-)

And RE: "optimization". I still don't get why it's wrong. It is a correct implementation according to the spec:
"""
When Read encounters an error or end-of-file condition after successfully reading n > 0 bytes, it returns the number of bytes read. It may return the (non-nil) error from the same call or return the error (and n == 0) from a subsequent call. An instance of this general case is that a Reader returning a non-zero number of bytes at the end of the input stream may return either err == EOF or err == nil. The next Read should return 0, EOF.
"""
 Show all checks